tss-esapi/src/structures/tagged/signature.rs

```diff
@@ -0,0 +1,125 @@
+use crate::{
+    interface_types::algorithm::SignatureScheme,
+    structures::{EccSignature, HashAgile, RsaSignature},
+    tss2_esys::{TPMT_SIGNATURE, TPMU_SIGNATURE},
+    Error, Result,
+};
+use std::convert::{TryFrom, TryInto};
+
+/// Enum representing a Signature
+///
+/// # Details
+/// This corresponds TPMT_SIGNATURE
+#[derive(Debug, Clone)]
+pub enum Signature {
+    RsaSsa(RsaSignature),
+    RsaPss(RsaSignature),
+    EcDsa(EccSignature),
+    EcDaa(EccSignature),
+    Sm2(EccSignature),
+    EcSchnorr(EccSignature),
+    Hmac(HashAgile),
+    Null,
+}
+
+impl Signature {
+    pub const fn algorithm(&self) -> SignatureScheme {
```

Comment:
Probably better to call it `scheme` (?)

Reply:
Yeah, I think you are right. Will rename the scheme and the algorithm.